src/desktop/ApplicationWindow.js

```diff
@@ -363,13 +364,13 @@ export class ApplicationWindow {
 		return this._browserWindow.webContents.getURL().substring(this._startFileURLString.length)
 	}
 
-	findInPage([searchTerm, options]: [string, {forward: boolean, matchCase: boolean}]): Promise<FindInPageResult> {
+	findInPage([searchTerm, options]: [string, {forward: boolean, matchCase: boolean}]): Promise<?FindInPageResult> {
 		this._findingInPage = true
 		if (searchTerm !== '') {
 			this._lastSearchRequest = [searchTerm, options]
 			this._browserWindow.webContents.findInPage(searchTerm, options)
 			return new Promise((resolve, reject) => {
-				this._lastSearchPromiseReject("outdated request")
+				this._lastSearchPromiseReject(new CancelledError("search request was superseded"))
 				this._lastSearchPromiseReject = reject
```

Comment:
is this correct setting `_lastSearchPromiseReject` twice here?

Reply:
_lstSearchPromiseReject is a function
the first line calls the last value with an error (rejecting the last search request if it is still ongoing, doing nothing if  the request is done because the function is replaced with noOp on request completion)

the second line replaces the function with the new request's reject callback.